src/rivian/ble.py

```diff
@@ -0,0 +1,68 @@
+import platform
+import asyncio
+import secrets
+import logging
+
+from bleak import BleakClient, BLEDevice
+from rivian import utils
+
+_LOGGER = logging.getLogger(__name__)
+
+PHONE_ID_VEHICLE_ID_UUID = "AA49565A-4D4F-424B-4559-5F5752495445"
+PNONCE_VNONCE_UUID = "E020A15D-E730-4B2C-908B-51DAF9D41E19"
+BLE_ACTIVE_ENTRY_UUID = "5249565F-4D4F-424B-4559-5F5752495445"
+
+async def pair_phone(
+    device: BLEDevice, vehicle_id, phone_id, vehicle_key, private_key
+) -> bool:
+    success = True
+
+    # Create an asyncio.Event object to signal the arrival of a new notification.
+    notification_event = asyncio.Event()
+    notification_data = None
+
+    # Callback for notifications from vehicle characteristics 
+    def notification_handler(_, data: bytearray):
+        nonlocal notification_data
+        notification_data = data
+        notification_event.set()
+
+    try:
+        async with BleakClient(device, timeout=10) as client:
+            _LOGGER.debug(f"Connecting to {BLEDevice.name} [{BLEDevice.address}]")
+            await client.start_notify(PHONE_ID_VEHICLE_ID_UUID, notification_handler)
+            await client.start_notify(PNONCE_VNONCE_UUID, notification_handler)
```

Comment:
We should either wait to start the notification on this characteristic or modify the notification_handler to check the characteristic uuid to ensure we got back the appropriate type of message.